### ISSUE TITLE
Rule for `bind` inside JSX

### ISSUE THREAD

**Issue by nkt** [yannickcr/eslint-plugin-react#184]
Is that possible to create rule that would throw warning on this code:
```jsx
<a onClick={this.onClick.bind(this, true)}>click me</a>
```
And maybe suggest to do this:
```jsx
<a onClick={() => this.onClick(true)}>click me</a>
```

**Comment by tleunen** [comment]
But your suggestion is not better... It's actually pretty much the same thing.

A good alternative is to bind the function inside the constructor, or to use babel stage 0 and bind it like this:
```
class A {
   myBoundFunction = () => {
   }
}
```
Or use the autobind decorator: https://github.com/jayphelps/core-decorators.js#autobind

But I think a rule for this is a good idea.

**Comment by Daniel15** [comment]
Yeah, either of those are not ideal as they both create a new bound function on every single `render` call. Garbage collectors don't like that. I'd say that both of those patterns should be caught by a lint rule :stuck_out_tongue: 

Maybe I'm old-school but I don't mind manually binding in the constructor (`this._onClick = this._onClick.bind(this)`). The autobind decorator @tleunen  linked to looks pretty fancy though.

**Comment by jhabdas** [comment]
For those having problems using `core-decorators` with Babel 6 you [may still use the `autobind` decorator](technologyadvice.github.io/es7-decorators-babel6/). I do not recommend using anything out of the Strawman proposal (Stage 0)
